textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -118,7 +126,14 @@ public static ITuple getSpanTuple( List<IField> fieldList, List<Span> spanList,
      */
     public static Schema createSpanSchema(Schema schema) {
         List<Attribute> dataTupleAttributes = schema.getAttributes();
-        //spanAttributes contains all attributes of dataTupleAttributes and an additional SPAN_LIST_ATTRIBUTE
+        // If SourceOperator is of type Keyword, spanschema is already provided

```

Comment:
This is related to the previous comment. In this method, we are creating a new schema with span details also included. When we use the KeyWord Source Operator type, the keywordMatcher returns a tuple to dictionaryMatcher with the span already included. So the span schema is already present, and now when we ignore the KeywordMatcher's span details and create the new span details. we don't have to create a new schema again because that has already been done by keywordMatcher.